react/react-ecosystem/type-checking-with-proptypes.md

```diff
@@ -0,0 +1,100 @@
+### Introduction
+
+PropTypes is a way to type check the props that a React component receives. It helps to catch potential type errors during development, making it easier to spot and fix bugs. If you have used a linter in your previous React projects, there's a good chance it ended up yelling at you about certain props missing in prop validation, however if that isn't the case- well, lucky you!
```

Comment:
We should start the introduction with a little explanation of what "type-checking"is. Afterwards we can introduce "PropTypes is a way ..."